packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -61,6 +61,7 @@ const ProductDetail = () => {
     if (res) {
         res.set('Cache-Control', `max-age=${MAX_CACHE_AGE}`)
     }
+    const isBasketLoading = !basket?.basketId
```

Comment:
(not important) you could use the query state `isLoading`. if in the future that we want to move to basket creation on demand, this won't work.

```
const {data: basket, isLoading: isBasketLoading} = useCurrentBasket()
```

Reply:
I tried this. The isLoading property state from useCurrentBasket isn't quite what we want since it's possible for isLoading to be false when the current basket is {} rather than undefined.